security/ssh_certificate.ksy

```diff
@@ -0,0 +1,162 @@
+      - id: type
+        type: u4
+        enum: cert_type
+        doc: specifies the certificate type (user vs host)
+      - id: key_id
+        type: cstring_utf8
+      - id: valid_principals
+        type: packed_cstring
+        doc: |
+          As a special case, a zero-length "valid principals" field means
+          the certificate is valid for any principal of the specified type.
+      - id: valid_after
+        type: u8
+        doc: Unix timestamp (seconds since 1970-01-01 00:00:00)
+      - id: valid_before
+        type: u8
+        doc: |
+          Unix timestamp (seconds since 1970-01-01 00:00:00)
+          When "forever" is requested this is set to 0xFFFF_FFFF_FFFF_FFFF
+      - id: critical_options
+        type: packed_cstring_tuple
+        doc: |
+          Contains zero or more options that are considered "critical".
+          They are considered "critical" as implementations must refuse
+          to authorize a certificate that has unrecognized options. This
+          prevents an unknown restriction in the certificate from failing to
+          be applied.
+      - id: extensions
+        type: packed_cstring_tuple
+        doc: |
+          Contains zero or more optional extensions. These extensions
+          are not critical, and an implementation that encounters one
+          that it does not recognize may safely ignore it.
+      - id: reserved
+        type: cstring_bytes
+        doc: Unused currently
+      - id: signature_key
+        type: cstring_sshkey
+        doc: |
+          The signature key field contains the CA key used to sign the
+          certificate. The valid key types for CA keys are ssh-rsa,
+          ssh-dss, ssh-ed25519 and the ECDSA types ecdsa-sha2-nistp256,
+          ecdsa-sha2-nistp384, ecdsa-sha2-nistp521. "Chained" certificates, where
+          the signature key type is a certificate type itself are NOT supported.
+          Note that it is possible for a RSA certificate key to be signed by a
+          Ed25519 or ECDSA CA key and vice-versa.
+      - id: signature
+        type: packed_cstring_tuple
+        doc: |
+          signature is computed over all preceding fields from the initial string
+          up to, and including the signature key. Signatures are computed and
+          encoded according to the rules defined for the CA's public key algorithm
+          (RFC4253 section 6.6 for ssh-rsa and ssh-dss, RFC5656 for the ECDSA
+          types), and draft-josefsson-eddsa-ed25519-03 for Ed25519.
+    enums:
+      cert_type:
+        1: user
+        2: host
+  packed_cstring:
+    seq:
+      - id: len_packed_strings
+        type: u4
+      - id: packed_strings
+        size: len_packed_strings
+        type: cstrings
+  packed_cstring_tuple:
+    seq:
+      - id: len_packed_strings
+        type: u4
+      - id: packed_strings
+        type: cstring_tuples
+        size: len_packed_strings
```

Comment:
What about just

```suggestion
      - id: len_tuples
        type: u4
      - id: tuples
        type: cstring_tuples
        size: len_tuples
```

...? In other words, does the adjective **"packed"** give any information about the meaning of the value?